src/Cli/dotnet/commands/dotnet-add/dotnet-add-reference/Program.cs

```diff
@@ -17,42 +17,29 @@ namespace Microsoft.DotNet.Tools.Add.ProjectToProjectReference
 {
     internal class AddProjectToProjectReferenceCommand : CommandBase
     {
-        private readonly AppliedOption _appliedCommand;
         private readonly string _fileOrDirectory;
 
-        public AddProjectToProjectReferenceCommand(
-            AppliedOption appliedCommand, 
-            string fileOrDirectory,
-            ParseResult parseResult) : base(parseResult)
+        public AddProjectToProjectReferenceCommand(ParseResult parseResult) : base(parseResult)
         {
-            if (appliedCommand == null)
-            {
-                throw new ArgumentNullException(nameof(appliedCommand));
-            }
-            if (fileOrDirectory == null)
-            {
-                throw new ArgumentNullException(nameof(fileOrDirectory));
-            }
-
-            _appliedCommand = appliedCommand;
-            _fileOrDirectory = fileOrDirectory;
+            _fileOrDirectory = parseResult.ValueForArgument<string>(AddCommandParser.ProjectArgument);
```

Comment:
There are overloads for `ValueForArgument` and `ValueForOption` that take `Argument<T>` and `Option<T>` respectively, so if these objects, rather than just the strings, can be referenced here, it should provide some perf benefit because the lookups are more efficient.